java/src/main/java/ai/onnxruntime/MapInfo.java

```diff
@@ -42,6 +44,21 @@ public class MapInfo implements ValueInfo {
     this.valueType = valueType;
   }
 
+  /**
+   * Construct a MapInfo with the specified size, key type and value type.
+   *
+   * <p>Called from JNI.
+   *
+   * @param size The size.
+   * @param keyTypeInt The int representing the {@link OnnxTensorType} of the keys.
+   * @param valueTypeInt The int representing the {@link OnnxTensorType} of the values.
+   */
+  MapInfo(int size, int keyTypeInt, int valueTypeInt) {
+    this.size = size;
+    this.keyType = OnnxJavaType.mapFromOnnxTensorType(OnnxTensorType.mapFromInt(keyTypeInt));
+    this.valueType = OnnxJavaType.mapFromOnnxTensorType(OnnxTensorType.mapFromInt(valueTypeInt));
+  }
+
   @Override
   public String toString() {
     String initial = size == -1 ? "MapInfo(size=UNKNOWN" : "MapInfo(size=" + size;
```

Comment:
>size == -1 [](http://example.com/codeflow?start=21&length=10)

How is the user to interpret the size of -1? Like I read size and it is -1, what am I going to do?

Reply:
It's -1 when the info is generated by `session.getOutputInfo` as we don't know the size of the map without executing the model. At all other times the size is set to whatever the map size is.

Reply:
Well, it is a piece of the metadata. ONNX definition does not include any length, it simply does not exist.

Reply:
Sure, but this class is two things, the first is a description of the model's output that we know a priori, and the second is the description of a specific ONNX map object. In the latter case I think it's useful to report the size in the toString.